extensions/graphql/README.md

```diff
@@ -0,0 +1,245 @@
+# @loopback/graphql
+
+This module provides integration with [GraphQL](https://graphql.org/) using
+[type-graphql](https://typegraphql.com/).
+
+![type-graphql](type-graphql.png)
+
+## Stability: ⚠️Experimental⚠️
+
+> Experimental packages provide early access to advanced or experimental
+> functionality to get community feedback. Such modules are published to npm
+> using `0.x.y` versions. Their APIs and functionality may be subject to
+> breaking changes in future releases.
+
+## Installation
+
+```sh
+npm install --save @loopback/graphql
+```
+
+## Basic Use
+
+```ts
+export class MyApplication extends BootMixin(RestApplication) {
+  constructor(config: ApplicationConfig) {
+    super(config);
+    this.projectRoot = __dirname;
+    this.component(GraphQLComponent);
+    this.configure(GraphQLBindings.GRAPHQL_SERVER).to({asMiddlewareOnly: true});
+  }
+}
+```
+
+## Configure GraphQLServer
+
+This package can be used in two flavors:
+
+- As a server for LoopBack applications
+
+```ts
+import {Application} from '@loopback/core';
+import {GraphQLServer} from '@loopback/graphql';
+
+const app = new Application();
+const serverBinding = app.server(GraphQLServer);
+app.configure(serverBinding.key).to({host: '127.0.0.1', port: 0});
+server = await app.getServer(GraphQLServer);
+// ...
+await app.start();
+```
+
+- As a middleware for LoopBack REST applications
+
+```ts
+import {BootMixin} from '@loopback/boot';
+import {RepositoryMixin} from '@loopback/repository';
+import {RestApplication} from '@loopback/rest';
+import {ApplicationConfig} from '@loopback/core';
+import {GraphQLComponent, GraphQLBindings} from '@loopback/graphql';
+
+export class GraphqlTestApplication extends BootMixin(
+  RepositoryMixin(RestApplication),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+
+    this.component(GraphQLComponent);
+    const server = this.getSync(GraphQLBindings.GRAPHQL_SERVER);
+    this.expressMiddleware('middleware.express.GraphQL', server.expressApp);
+    this.configure(GraphQLBindings.GRAPHQL_SERVER).to({
+      asMiddlewareOnly: true,
+    });
+
+    // ...
+
+    // Customize @loopback/boot Booter Conventions here
+    this.bootOptions = {
+      graphqlResolvers: {
+        // Customize ControllerBooter Conventions here
+        dirs: ['graphql-resolvers'],
+        extensions: ['.js'],
+        nested: true,
+      },
+    };
+  }
+}
+```
+
+## Add a resolver
+
+Please add `recipe-resolver.ts` to `src/graphql-resolvers`.
+
```

Comment:
As a person new to GraphQL and LoopBack, I would find it difficult to understand what's going on in the code snippet below. It would be nice to provide a brief explanation of concepts here.

1. Individual query/mutation/resolver functions are grouped to resolver classes, similarly to how REST API endpoints are grouped to controller classes.

2. We use TypeScript decorators to provide metadata about individual resolvers. 

3. Explain what is the purpose (effect) of the class-level `@resolver`, method-level `@query`, `@mutation` and `@fieldResolver` decorators.

You may want to consider adding a link to type-graphql documentation too, to provide readers with resources where they can learn more.